### PR TITLE
Fix Character Table, Mask of GCNN

### PR DESCRIPTION
**Pull request by chrisrothUT**
Really sorry to open up another pull request, but I found a few more bugs.

### CODE REVIEW

File: netket/nn/symmetric_linear.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def __call__(self, x: Array) -> Array:
             "kernel", self.kernel_init, (self.features, self.n_sites), self.dtype
         )
 
-        if self.mask:
+        if not self.mask is None:
```
Comment (PhilipVinc) [inline netket/nn/symmetric_linear.py:129]: python would like you to use
```suggestion
        if self.mask is not None:
```

Comment (chrisrothUT) [reply netket/nn/symmetric_linear.py:129]: Okay, that's in four places so I'll fix it on my end

**Comment by PhilipVinc** [comment]
could you just add a `very quick test` to check that this does not happen again?

**Comment by PhilipVinc** [comment]
Thanks.
though I think we don't need to check for all possible combinations that the output is also correct, 
I mean, in principle we want to, but it's going to take a serious while... it's a combinatorial explosion of combinations of the options.
Our CI Starts to take really long.

ideally for some options we'd just check that they don't crash (which I guess is what was happening)

**Comment by PhilipVinc** [comment]
@chrisrothUT can you run `flakehell lint netket/models` and `flakehell lint netket/nn` and fix the issues raised please?

**Comment by PhilipVinc** [comment]
@gcarleo this is now functional. Beware that it only works for `vstate.expect(operator)`

To play with it, you should set `vstate.minibatch_size = XXX` *after* you create the model.
whatever value you input it is rounded to the next multiple of `sampler.n_chains`.
Also, the number of samples might be increased so that it's a multiple of `minibatch_size`.

We might want to fine tune this logic later on.

This PR has a runtime cost for users that don't want to use minibatches: `~50μs` per call to `expect(op)`, Which is mainly due to an extra call to a jax-jitted function . 
We might get rid of this cost but the implementation would be more complicated, and `50μs` is... not much, considering that the Ising example takes `~20-50ms` per iteration on my computer

**Comment by chrisrothUT** [comment]
Okay, I fixed the issues that were raised

**Review by PhilipVinc — APPROVED** [review]

**Pull request merged**